sudokuGUI.py

```diff
@@ -1,532 +1,533 @@
+        solving(self.finished_board)
+        self.load_board()
+        self.state = 'playing'
+        self.start = time.time()
+
+# Playing state functions # 
+
+    # Function that responsible on the playing differen events
+    # Input: Nothing
+    # Output: Nothing
+    def playing_events(self):
+        for event in pygame.event.get():
+            if event.type == pygame.QUIT:
+                self.running = False
+            
+            # When the user clicks
+            if event.type == pygame.MOUSEBUTTONDOWN:
+                if self.mouse_pos:
+                    self.error_msg = False
+                    selected = self.mouse_on_grid()
+
+                if selected:
+                    self.selected = selected
+                else:
+                    self.selected = None
+                    for button in self.playing_buttons:
+                        if button.highlighted:
+                            button.click()
+                            break
+            
+            # When the user type
+            if event.type == pygame.KEYDOWN:
+                if self.selected != None and self.selected not in self.lock_cells: # Checking if the position is a blank rect
+                    if self.is_int(event.unicode): # Checking if the user entered a number 
+                        self.temp_values[(self.selected[0], self.selected[1])] =  int(event.unicode) # temp_values[(x, y)] = num
+                    
+                    elif event.key == pygame.K_BACKSPACE:
+                        del self.temp_values[(self.selected[0], self.selected[1])]
+
+                if event.key == pygame.K_KP_ENTER:
+                    pos = (self.selected[0], self.selected[1])
+                    if self.finished_board[self.selected[1]][self.selected[0]] == self.temp_values[pos]: # Checking if the entered number is correct
+                        self.grid[self.selected[1]][self.selected[0]] = self.temp_values[pos] # Adding to the board
+                        del self.temp_values[pos] # Delete the value from the temp_values list -> value became permenent
+                        self.cell_changed = True
+                    else: # Otherwise it is an error
+                        self.errors += 1
+
+    
+    # Function that updates the game (by user's mouse position/clicking on buttons etc)
+    # Input: Nothing
+    # Output: Nothing
+    def playing_update(self):
+        self.mouse_pos = pygame.mouse.get_pos()
+        for button in self.playing_buttons:
+            button.update(self.mouse_pos)
+        
+        if self.cell_changed:
+            if empty_pos(self.grid) is None:
+                self.state = 'win'
+                self.screen_text = 'You Win'
+                self.caption = 'Sudoku - Win screen'
+                self.game_over_draw()   
+                
+                
+    # Function that draw the game
+    # Input: Nothing
+    # Output: Nothing
+    def playing_draw(self):
```

Comment:
This function is very long, try to see if you can separate it to smaller functions